Marlin/Marlin_main.cpp

```diff
@@ -945,38 +949,221 @@ void servo_init() {
+        #endif
+      #endif  //BLINKM
+  }
+  #endif  //!LEDSTRIP
+
+    /*  Handle the various printer events
+   *  
+   *  0 - Green       - Timed or click to OFF.
+   *  1 - White       - Used for main printing as a case light.
+   *  2 - Yellow      - Used when homing.
+   *  3 - Purple      - Not currently used.
+   *  4 - Aqua        - Used for filament change.
+   *  5 - Aqua dimmed - Used for filment change hotend timed out.
+   *  6 - Aqua Half   - Used for filament change hotend heating.
+   *  7 - Red         - Not currently used.
+   *  9 - Off         - Used for after homing and leveling.
+   */
 
-      // This variant uses 3 separate pins for the RGB components.
-      // If the pins can do PWM then their intensity will be set.
-      digitalWrite(RGB_LED_R_PIN, r ? HIGH : LOW);
-      digitalWrite(RGB_LED_G_PIN, g ? HIGH : LOW);
-      digitalWrite(RGB_LED_B_PIN, b ? HIGH : LOW);
-      analogWrite(RGB_LED_R_PIN, r);
-      analogWrite(RGB_LED_G_PIN, g);
-      analogWrite(RGB_LED_B_PIN, b);
+  // Handle the various printer events
+  void handle_led_print_event(uint8_t code) {
+    uint16_t wait_for_user_timeout = 0;
+    switch(code) {
+      case(0):        // Print Complete
+        LCD_MESSAGEPGM(MSG_INFO_COMPLETED_PRINTS);
+        lcd_update();
+        #if ENABLED(LEDSTRIP)
+          SendColorsOnLedstrip (0, 255, 0, 0, 1);  // Turn LEDs Green
+        #else
+          set_led_color(0, 255, 0, 0);  // Turn RGB LEDs to GREEN
+        #endif
 
-      #if ENABLED(RGBW_LED)
-        digitalWrite(RGB_LED_W_PIN, w ? HIGH : LOW);
-        analogWrite(RGB_LED_W_PIN, w);
-      #endif
+        #if DISABLED(NO_PAUSE_OR_TIMEOUT)
+          wait_for_user = true;
+          do {
+            idle();
+            safe_delay(100);
+            if (wait_for_user_timeout == (LED_reset_time * 5))
+            #if ENABLED(ULTRA_LCD) || ENABLED(DOGLCD)
+              SERIAL_ECHOLNPGM(MSG_BUSY_PAUSED_FOR_USER_OR_TIMEOUT);
+            #endif  
+            wait_for_user_timeout ++;
+            if (wait_for_user_timeout >= (LED_reset_time * 10)) break;
+          } while (wait_for_user);
+          wait_for_user = false;
+          #if ENABLED(LEDSTRIP)
+            SendColorsOnLedstrip (0, 0, 0, 0, 0);  // Turn RGB LEDs off
+          #else
+            set_led_color(0, 0, 0, 0);  // Turn RGB LEDs off
+          #endif
+        #endif  // NO_PAUSE_OR_TIMEOUT
 
-    #endif
-  }
+        LCD_MESSAGEPGM(WELCOME_MSG);
+        idle();
+        break;
+      case(1):      // Turn RGB LEDs White
+        #if ENABLED(LEDSTRIP)
+          SendColorsOnLedstrip (255, 255, 255, 0, 1);
+        #elif ENABLED(RGBW_STRIP)
+          set_led_color(0, 0, 0, 255);
+        #else
+          set_led_color(255, 255, 255, 0);
+        #endif
+        break;
+      case(2):      // Turn RGB LEDs Yellow
+        #if ENABLED(LEDSTRIP)
+          SendColorsOnLedstrip (255, 255, 0, 0, 1);
+        #else
+          set_led_color(255, 255, 0, 0);
+        #endif
```

Comment:
Instead of repeating many of these blocks you should do this up above the function:
```cpp
#if ENABLED(LEDSTRIP)
  #define LED_COLOR_RGBW(R,G,B,W) SendColorsOnLedstrip(R, G, B, W, 1)
#else
  #define LED_COLOR_RGBW(R,G,B,W) set_led_color(R, G, B, W)
#endif
#define LED_COLOR_RGB(R,G,B) SET_COLOR_RGBW(R,G,B,0)
```
Then you can simply use this form:
```cpp
LED_COLOR_RGB(255, 255, 0);
```
…and/or name your function `set_led_color` instead.